### PR TITLE
fix: readFileNode return string | null

### PR DESCRIPTION
**Pull request by yjhmelody**
The d.ts:

```ts
 /** Reads a file from disk (or memory). */
  readFile?: (filename: string, baseDir: string) => string | null;
```

### CODE REVIEW

File: cli/asc.js [hunk 1]
```diff
@@ -1177,8 +1177,8 @@ exports.main = function main(argv, options, callback) {
   }
 
   function listFilesNode(dirname, baseDir) {
-    var files;
```
Comment (MaxGraey) [inline cli/asc.js:1180]: Hmm, why this change necessary?

**Comment by MaxGraey** [comment]
I'm not sure is this fix something but it probably better use following approach:
```ts
let text = null;
try {
  text = /* I/O read */
} finally {
  return text;
}
```
instead try-catch

**Comment by yjhmelody** [comment]
> I'm not sure is this fix something but it probably better use following approach:
> 
> ```ts
> let text = null;
> try {
>   text = /* sync I/O read */
> } finally {
>   return text;
> }
> ```
> 
> instead try-catch

eslint error:
Unsafe usage of ReturnStatement.eslintno-unsafe-finally

**Comment by MaxGraey** [comment]
I see. But I'm wondering why [here](https://github.com/AssemblyScript/assemblyscript/pull/1830/files#diff-609c82113feed36cb75ab2bd3d387762bf2829fc5ae10477722fee8a97885489R1152) you add `null`? it should return null only when exception occurs. Otherwise `text` overwriting by `fs.readFileSync` with non-null value.

**Comment by yjhmelody** [comment]
Sorry for the wrong understanding, this function should be no problem. I closed it.
